File: src/HotelOrderSystem.Api/Entities/Order.cs
using System.ComponentModel.DataAnnotations;

namespace HotelOrderSystem.Api.Entities;

public sealed class Order
{
    public int OrderId { get; set; }
    public int RoomId { get; set; }
    public Room Room { get; set; } = null!;

    public int? CreatedByUserId { get; set; }
    public User? CreatedByUser { get; set; }

    public int? AssignedTeamId { get; set; }
    public Team? AssignedTeam { get; set; }

    public string Source { get; set; } = "Admin";
    public string Status { get; set; } = "Pending";

    public int? AcceptedByUserId { get; set; }
    public User? AcceptedByUser { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? AcceptedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? CancelledAt { get; set; }
    public DateTime? SlaDueAt { get; set; }
    public DateTime? EscalatedAt { get; set; }

    [Timestamp]
    public byte[] RowVersion { get; set; } = Array.Empty<byte>();

    public ICollection<OrderDetail> Details { get; set; } = new List<OrderDetail>();
}

File: android-staff-app/app/src/main/java/com/ibaapps/HotelOrderSystem/domain/model/Order.kt
package com.ibaapps.HotelOrderSystem.domain.model

import java.time.Instant

enum class OrderStatus {
    Pending,
    Accepted,
    InProgress,
    Completed,
    Cancelled,
    Unknown;

    companion object {
        fun fromRaw(raw: String?): OrderStatus = when (raw?.trim()?.lowercase()) {
            "pending" -> Pending
            "accepted" -> Accepted
            "inprogress", "in_progress", "in progress" -> InProgress
            "completed" -> Completed
            "cancelled", "canceled" -> Cancelled
            else -> Unknown
        }
    }
}

data class Order(
    val orderId: Int,
    val roomId: Int,
    val roomNumber: String,
    val assignedTeamId: Int?,
    val assignedTeamName: String?,
    val source: String,
    val status: OrderStatus,
    val statusRaw: String,
    val createdByUserName: String?,
    val acceptedByUserId: Int?,
    val acceptedByUserName: String?,
    val createdAt: Instant?,
    val acceptedAt: Instant?,
    val completedAt: Instant?,
    val slaDueAt: Instant?,
    val escalatedAt: Instant?,
    val rowVersion: String,
    val details: List<OrderLine>
) {
    /** Primary item/service name for compact cards (first line, or a count summary). */
    val primaryItemName: String
        get() = when {
            details.isEmpty() -> "Request"
            details.size == 1 -> details.first().itemName
            else -> "${details.first().itemName} +${details.size - 1}"
        }

    val isEscalated: Boolean get() = escalatedAt != null
}

data class OrderLine(
    val orderDetailId: Int,
    val itemId: Int,
    val itemName: String,
    val quantity: Int,
    val dynamicAttributes: Map<String, String>
)
